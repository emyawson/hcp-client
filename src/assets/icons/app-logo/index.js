export * from './app-logo.component';
export * from './app-logo-wordmark.component';
