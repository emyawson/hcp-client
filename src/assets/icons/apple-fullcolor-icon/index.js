export * from './apple-fullcolor-icon.component';
