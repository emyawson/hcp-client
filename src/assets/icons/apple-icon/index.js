export * from './apple-icon.component';
