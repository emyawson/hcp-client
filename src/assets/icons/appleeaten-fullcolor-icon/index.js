export * from './appleeaten-fullcolor-icon.component';
