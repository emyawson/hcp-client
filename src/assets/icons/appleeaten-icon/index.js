export * from './appleeaten-icon.component';
