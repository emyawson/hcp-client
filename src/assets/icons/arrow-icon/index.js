export * from './arrow.component';
export * from './arrow-right.component';
