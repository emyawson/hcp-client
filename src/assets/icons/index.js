export * from './app-logo';
export * from './apple-fullcolor-icon';
export * from './apple-icon';
export * from './appleeaten-fullcolor-icon';
export * from './appleeaten-icon';
export * from './arrow-icon';
export * from './asterisk-icon';
export * from './basal-plus-bolus-icon';
export * from './bell-icon';
export * from './bolus-extended-icon';
export * from './bolus-multiwave-icon';
export * from './bolus-quick-icon';
export * from './bolus-standard-icon';
export * from './box-graph-icon';
export * from './ce-icon';
export * from './checkmark-icon';
export * from './circle-mark-icon';
export * from './connecting-lines-icon';
export * from './create-patient-icon';
export * from './cross-mark-icon';
export * from './date-range-icon';
export * from './disclaimer-icon';
export * from './double-ellipse-icon';
export * from './dtc-icon';
export * from './factory-icon';
export * from './flag-icon';
export * from './gear-icon';
export * from './grid-icon';
export * from './help-icon';
export * from './home-icon';
export * from './hypo-icon';
export * from './hypo-limit-icon';
export * from './hypo-symptoms-icon';
export * from './icon';
export * from './left-chevron-icon';
export * from './left-guillemet-icon';
export * from './light-bulb-icon';
export * from './line-graph-icon';
export {
  LogbookDiaryIcon,
  LogbookIcon,
  LogbookStatsIcon,
} from './logbook-icon';
export * from './logo';
export * from './maximize-icon';
export * from './mean-blood-glucose-icon';
export * from './medical-icon';
export * from './menu-icon';
export * from './minimize-icon';
export * from './minus-icon';
export * from './modified-bolus-icon';
export * from './night-icon';
export * from './no-information-icon';
export * from './overnight-icon';
export * from './patients-icon';
export * from './patient-dashboard-icon';
export * from './pencil-icon';
export * from './plus-icon';
export * from './pump-pause-icon';
export * from './pump-power-down-icon';
export * from './pump-power-up-icon';
export * from './pump-profile-change-icon';
export * from './pump-basal-rate-change-icon';
export * from './pump-run-icon';
export * from './pump-stop-icon';
export * from './pump-tbrdec-icon';
export * from './pump-tbrenddec-icon';
export * from './pump-tbrendinc-icon';
export * from './pump-tbrinc-icon';
export * from './question-icon';
export * from './rectangle-mark-icon';
export * from './right-chevron-icon';
export * from './right-guillemet-icon';
export * from './speech-bubble-icon';
export * from './stripstock-icon';
export * from './t-icon';
export * from './trash-icon';
export * from './target-range-icon';
export * from './white-line-icon';
export * from './x-icon';
export * from './patterns-icon';
export * from './device-icon';
export * from './device-error-icon';
export * from './device-assignment-error-icon';
export * from './device-download-icon';
export * from './patient-silhouette-icon';
export * from './phone-icon';
export * from './email-icon';
export * from './udtc-icon';
