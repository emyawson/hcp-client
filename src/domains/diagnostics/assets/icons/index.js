export { AppleIcon } from './apple-icon';
export { AppleEatenIcon } from './appleeaten-icon';
export { AppleEatenFullColorIcon } from './appleeaten-fullcolor-icon';
export { AppleFullColorIcon } from './apple-fullcolor-icon';
export { ArrowIcon } from './arrow-icon';
export { AsteriskIcon } from './asterisk-icon';
export { BasalPlusBolusIcon } from './basal-plus-bolus-icon';
export { BolusExtendedIcon } from './bolus-extended-icon';
export { BolusMultiwaveIcon } from './bolus-multiwave-icon';
export { BolusQuickIcon } from './bolus-quick-icon';
export { BolusStandardIcon } from './bolus-standard-icon';
export { BoxGraphIcon } from './box-graph-icon';
export { CheckmarkIcon } from './checkmark-icon';
export { CircleMarkIcon } from './circle-mark-icon';
export { ConnectingLinesIcon } from './connecting-lines-icon';
export { CrossMarkIcon } from './cross-mark-icon';
export { DateRangeIcon } from './date-range-icon';
export { DisclaimerIcon } from './disclaimer-icon';
export { DoubleEllipseIcon } from './double-ellipse-icon';
export { GearIcon } from './gear-icon';
export { GridIcon } from './grid-icon';
export { HypoIcon } from './hypo-icon';
export { HypoLimitIcon } from './hypo-limit-icon';
export { HypoSymptomsIcon } from './hypo-symptoms-icon';
export { LeftGuillemetIcon } from './left-guillemet-icon';
export { LeftChevronIcon } from './left-chevron-icon';
export { LightBulbIcon } from './light-bulb-icon';
export { LineGraphIcon } from './line-graph-icon';
export {
  LogbookIcon,
  LogbookDiaryIcon,
  LogbookStatsIcon,
} from './logbook-icon';
export { MinimizeIcon } from './minimize-icon';
export { MaximizeIcon } from './maximize-icon';
export { MeanBloodGlucoseIcon } from './mean-blood-glucose-icon';
export { MenuIcon } from './menu-icon';
export { ModifiedBolusIcon } from './modified-bolus-icon';
export { NightIcon } from './night-icon';
export { OvernightIcon } from './overnight-icon';
export { PumpBasalRateChangeIcon } from './pump-basal-rate-change-icon';
export { PumpPowerDownIcon } from './pump-power-down-icon';
export { PumpPowerUpIcon } from './pump-power-up-icon';
export { PumpProfileChangeIcon } from './pump-profile-change-icon';
export { PumpPauseIcon } from './pump-pause-icon';
export { PumpRunIcon } from './pump-run-icon';
export { PumpStopIcon } from './pump-stop-icon';
export { PumpTbrdecIcon } from './pump-tbrdec-icon';
export { PumpTbrEndDecIcon } from './pump-tbrenddec-icon';
export { PumpTbrincIcon } from './pump-tbrinc-icon';
export { PumpTbrEndIncIcon } from './pump-tbrendinc-icon';
export { RectangleMarkIcon } from './rectangle-mark-icon';
export { RightChevronIcon } from './right-chevron-icon';
export { RightGuillemetIcon } from './right-guillemet-icon';
export { TIcon } from './t-icon';
export { SnackIcon } from './snack-icon';
export { PillsIcon } from './pills-icon';
export { BedIcon } from './bed-icon';
export { RunningIcon } from './running-icon';
export { BicycleIcon } from './bicycle-icon';
export { MealIcon } from './meal-icon';
export { HyperWarningIcon } from './hyper-warning-icon';
export { HypoWarningIcon } from './hypo-warning-icon';
export { PremenstrualIcon } from './premenstrual-icon';
export { CustomIcon } from './custom-icon';
export { UserDefinedAsteriskIcon } from './user-defined-asterisk-icon';
export { StressIcon } from './stress-icon';
export { IllnessIcon } from './illness-icon';
export { FastingIcon } from './fasting-icon';
export {
  BeforeExerciseIcon,
  DuringExerciseIcon,
  AfterExerciseIcon,
} from './exercise-icon';
