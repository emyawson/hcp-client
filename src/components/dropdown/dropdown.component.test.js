import { shallow } from 'enzyme';
import React from 'react';

import { ArrowIcon } from 'src/assets/icons';

import {
  Dropdown,
  createValidators,
  Arrow,
  selectionHasValue,
} from './dropdown.component';

describe('dropdown test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(
      <Dropdown
        label="Drop down"
        placeholder="The placeholder"
        modelPath=".dropdown"
        options={[
          { label: 'Uno', value: 1 },
          { label: 'Dos', value: 2 },
          { label: 'Tres', value: 3 },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('creates RRF validators if the input is required', () => {
    expect(createValidators(true).isRequired).toBeDefined();
    expect(createValidators(false)).toEqual({});
  });
  it('validates selected option when required', () => {
    expect(selectionHasValue('test')).toBeTruthy();
    expect(selectionHasValue(null)).toBeFalsy();
    expect(selectionHasValue(1)).toBeTruthy();
    expect(selectionHasValue(undefined)).toBeFalsy();
  });
  it('should render an arrow with default styles if none are provided', () => {
    const wrapper = shallow(<Arrow />)
      .find(ArrowIcon)
      .first();
    expect(wrapper.props().fillColor).toBeDefined();
    expect(wrapper.props().height).toBeDefined();
  });
});
