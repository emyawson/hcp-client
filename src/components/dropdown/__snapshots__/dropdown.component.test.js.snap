// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    autosize={false}
    backspaceRemoves={false}
    clearable={false}
    disabled={false}
    id=""
    label="Drop down"
    modelPath=".dropdown"
    multi={false}
    onChange={[Function]}
    optionFontSize="0.875rem"
    options={
      Array [
        Object {
          "label": "Uno",
          "value": 1,
        },
        Object {
          "label": "Dos",
          "value": 2,
        },
        Object {
          "label": "Tres",
          "value": 3,
        },
      ]
    }
    placeholder="The placeholder"
    required={false}
    searchable={true}
    selectHeight="3rem"
    selectedOptionPadding="1rem"
    simpleValue={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "borderless": undefined,
      "children": <Unknown
        aria-label="Drop down"
        autosize={false}
        backspaceRemoves={false}
        clearable={false}
        component={[Function]}
        controlProps={
          Object {
            "arrowRenderer": [Function],
            "autosize": false,
            "backspaceRemoves": false,
            "borderless": undefined,
            "clearable": false,
            "disabled": false,
            "id": "",
            "menuContainerStyle": Object {},
            "menuStyle": Object {},
            "multi": false,
            "optionFontSize": "0.875rem",
            "options": Array [
              Object {
                "label": "Uno",
                "value": 1,
              },
              Object {
                "label": "Dos",
                "value": 2,
              },
              Object {
                "label": "Tres",
                "value": 3,
              },
            ],
            "required": false,
            "searchable": true,
            "selectHeight": "3rem",
            "selectedOptionPadding": "1rem",
            "simpleValue": true,
            "style": Object {},
          }
        }
        disabled={false}
        id=""
        model=".dropdown"
        multi={false}
        onChange={[Function]}
        optionFontSize="0.875rem"
        options={
          Array [
            Object {
              "label": "Uno",
              "value": 1,
            },
            Object {
              "label": "Dos",
              "value": 2,
            },
            Object {
              "label": "Tres",
              "value": 3,
            },
          ]
        }
        placeholder="The placeholder"
        searchable={true}
        selectHeight="3rem"
        selectedOptionPadding="1rem"
        simpleValue={true}
        validators={Object {}}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "aria-label": "Drop down",
        "autosize": false,
        "backspaceRemoves": false,
        "clearable": false,
        "component": [Function],
        "controlProps": Object {
          "arrowRenderer": [Function],
          "autosize": false,
          "backspaceRemoves": false,
          "borderless": undefined,
          "clearable": false,
          "disabled": false,
          "id": "",
          "menuContainerStyle": Object {},
          "menuStyle": Object {},
          "multi": false,
          "optionFontSize": "0.875rem",
          "options": Array [
            Object {
              "label": "Uno",
              "value": 1,
            },
            Object {
              "label": "Dos",
              "value": 2,
            },
            Object {
              "label": "Tres",
              "value": 3,
            },
          ],
          "required": false,
          "searchable": true,
          "selectHeight": "3rem",
          "selectedOptionPadding": "1rem",
          "simpleValue": true,
          "style": Object {},
        },
        "disabled": false,
        "id": "",
        "model": ".dropdown",
        "multi": false,
        "onChange": [Function],
        "optionFontSize": "0.875rem",
        "options": Array [
          Object {
            "label": "Uno",
            "value": 1,
          },
          Object {
            "label": "Dos",
            "value": 2,
          },
          Object {
            "label": "Tres",
            "value": 3,
          },
        ],
        "placeholder": "The placeholder",
        "searchable": true,
        "selectHeight": "3rem",
        "selectedOptionPadding": "1rem",
        "simpleValue": true,
        "validators": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "borderless": undefined,
        "children": <Unknown
          aria-label="Drop down"
          autosize={false}
          backspaceRemoves={false}
          clearable={false}
          component={[Function]}
          controlProps={
            Object {
              "arrowRenderer": [Function],
              "autosize": false,
              "backspaceRemoves": false,
              "borderless": undefined,
              "clearable": false,
              "disabled": false,
              "id": "",
              "menuContainerStyle": Object {},
              "menuStyle": Object {},
              "multi": false,
              "optionFontSize": "0.875rem",
              "options": Array [
                Object {
                  "label": "Uno",
                  "value": 1,
                },
                Object {
                  "label": "Dos",
                  "value": 2,
                },
                Object {
                  "label": "Tres",
                  "value": 3,
                },
              ],
              "required": false,
              "searchable": true,
              "selectHeight": "3rem",
              "selectedOptionPadding": "1rem",
              "simpleValue": true,
              "style": Object {},
            }
          }
          disabled={false}
          id=""
          model=".dropdown"
          multi={false}
          onChange={[Function]}
          optionFontSize="0.875rem"
          options={
            Array [
              Object {
                "label": "Uno",
                "value": 1,
              },
              Object {
                "label": "Dos",
                "value": 2,
              },
              Object {
                "label": "Tres",
                "value": 3,
              },
            ]
          }
          placeholder="The placeholder"
          searchable={true}
          selectHeight="3rem"
          selectedOptionPadding="1rem"
          simpleValue={true}
          validators={Object {}}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "aria-label": "Drop down",
          "autosize": false,
          "backspaceRemoves": false,
          "clearable": false,
          "component": [Function],
          "controlProps": Object {
            "arrowRenderer": [Function],
            "autosize": false,
            "backspaceRemoves": false,
            "borderless": undefined,
            "clearable": false,
            "disabled": false,
            "id": "",
            "menuContainerStyle": Object {},
            "menuStyle": Object {},
            "multi": false,
            "optionFontSize": "0.875rem",
            "options": Array [
              Object {
                "label": "Uno",
                "value": 1,
              },
              Object {
                "label": "Dos",
                "value": 2,
              },
              Object {
                "label": "Tres",
                "value": 3,
              },
            ],
            "required": false,
            "searchable": true,
            "selectHeight": "3rem",
            "selectedOptionPadding": "1rem",
            "simpleValue": true,
            "style": Object {},
          },
          "disabled": false,
          "id": "",
          "model": ".dropdown",
          "multi": false,
          "onChange": [Function],
          "optionFontSize": "0.875rem",
          "options": Array [
            Object {
              "label": "Uno",
              "value": 1,
            },
            Object {
              "label": "Dos",
              "value": 2,
            },
            Object {
              "label": "Tres",
              "value": 3,
            },
          ],
          "placeholder": "The placeholder",
          "searchable": true,
          "selectHeight": "3rem",
          "selectedOptionPadding": "1rem",
          "simpleValue": true,
          "validators": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
